code/modules/spells/spell_types/devil.dm

```diff
@@ -118,27 +120,55 @@
 				revert_cast()
 				return ..()
 		else
-			user.notransform = TRUE
-			user.fakefire()
-			to_chat(src, "<span class='warning'>You begin to phase back into sinful flames.</span>")
-			if(do_mob(user,user,150))
-				user.infernalphaseout()
+			if (satin && satin.can_jaunt)
+				if (!user.stat && !satin.check_banishment(user))
+					to_chat(user, "<span class='warning'>You have been banished from using your powers.</span>")
+					return FALSE
+				user.notransform = TRUE
+				user.fakefire()
+				if(do_mob(user,user,3 SECONDS))
+					to_chat(src, "<span class='warning'>You phase back into sinful flames.</span>")
+					satin.set_jaunt(FALSE)
+					user.infernalphaseout()
+				else
+					user.notransform = FALSE
+					user.fakefireextinguish()
 			else
-				to_chat(user, "<span class='warning'>You must remain still while exiting.</span>")
-				user.notransform = FALSE
-				user.fakefireextinguish()
+				to_chat(user, "<span class='warning'>You must sign a contract before you can jaunt again.</span>")
 		start_recharge()
 		return
 	revert_cast()
 
 /obj/effect/proc_holder/spell/targeted/infernal_jaunt/proc/valid_location(mob/living/user = usr)
-	if(istype(get_area(user), /area/shuttle/)) // Can always phase in in a shuttle.
+	if (!user.mind)
 		return TRUE
-	else
-		for(var/mob/living/C in orange(2, get_turf(user))) //Can also phase in when nearby a potential buyer.
-			if (C.owns_soul())
-				return TRUE
-	return FALSE
+	var/datum/antagonist/devil/satin = user.mind.has_antag_datum(/datum/antagonist/devil)
+	var/pick_range = 4
```

Comment:
Does this need to be a var? It can't be anything else except 4 and is only used in the switch below